src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -618,8 +632,7 @@ private ProcessingBranch PrepareProcessingBranch(Block suggestedBlock, Processin
                 // we need to dig deeper to go all the way to the false (reorg boundary) head
                 // otherwise some nodes would be missing
                 bool notFoundTheBranchingPointYet = !_blockTree.IsMainChain(branchingPoint.Hash!);
-                bool notReachedTheReorgBoundary = (options & ProcessingOptions.Trace) == ProcessingOptions.Trace
-                && (branchingPoint.Number > (_blockTree.Head?.Header.Number ?? 0));
+                bool notReachedTheReorgBoundary = branchingPoint.Number > (_blockTree.Head?.Header.Number ?? 0);
```

Comment:
Is this conscious change? This was just changed in https://github.com/NethermindEth/nethermind/pull/5090

Reply:
Yea, I think @OlegJakushkin said that @MarekM25 said that this is needed?

Reply:
@asdacap @OlegJakushkin @LukaszRozmej  just to be clear, I proposed this change as the temp hack to see if it resolves our memory problems with `trace_block`

Reply:
and this condition is wrong. I didn't want to check reachedTheReorgBoundry for traces. So like this:
` bool notReachedTheReorgBoundary = (options & ProcessingOptions.Trace) != ProcessingOptions.Trace
&& (branchingPoint.Number > (_blockTree.Head?.Header.Number ?? 0));`

not like this:
` bool notReachedTheReorgBoundary = (options & ProcessingOptions.Trace) == ProcessingOptions.Trace
&& (branchingPoint.Number > (_blockTree.Head?.Header.Number ?? 0));`

Reply:
@LukaszRozmej This was not needed to fix trace block at all as there were other checks more, global. Will remove it in next PR on this topic later today.

Reply:
@OlegJakushkin Was it improved on any other PR?